ragna/assistants/_bedrock.py

```diff
@@ -0,0 +1,79 @@
+from typing import cast
+import boto3
+import json
+from ragna.core import RagnaException, Source
+
+from ._api import ApiAssistant
+
+
+class AmazonBedrockAssistant(ApiAssistant):
+    _API_KEY_ENV_VAR = "AWS_ACCESS_KEY_ID"
+    _MODEL: str
+    _CONTEXT_SIZE: int
+
+    @classmethod
+    def display_name(cls) -> str:
+        return f"AmazonBedRock/{cls._MODEL}"
+
+    @property
+    def max_input_size(self) -> int:
+        return self._CONTEXT_SIZE
+
+    def _instructize_prompt(self, prompt: str, sources: list[Source]) -> str:
+        instruction = (
+            "\n\nHuman: "
+            "Use the following pieces of context to answer the question at the end. "
+            "If you don't know the answer, just say so. Don't try to make up an answer.\n"
+        )
+
+        instruction += "\n\n".join(source.content for source in sources)
+        return f"{instruction}\n\nQuestion: {prompt}\n\nAssistant:"
```

Comment:
This prompt is special for Anthropic, although it should also be working for other assistants.